cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/TransformDetail.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.common;
+
+import co.cask.cdap.api.metrics.Metrics;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.InvalidEntry;
+import co.cask.cdap.etl.api.Transformation;
+
+import java.util.List;
+
+/**
+ * Encapsulates {@link Transformation} list of next stages, current stage name and {@link DefaultEmitter}
+ */
+public class TransformDetail implements Emitter<Object> {
+  private final String stageName;
+  private final Transformation transformation;
+  private final List<String> nextStages;
+  private final DefaultEmitter defaultEmitter;
+
+  public TransformDetail(String stageName, Transformation transformation, Metrics metrics,
+                         List<String> nextStages, DefaultEmitter defaultEmitter) {
+    this.stageName = stageName;
+    this.transformation = transformation == null ? null :
```

Comment:
I see this is called in ETLWorker.  If there is no transformation it shouldn't be passed to the transform executor.  Should instead pass in the list of first stages to the executor rather than having a fake transform for that purpose.